### PR TITLE
Make the Personalization’s dynamicTemplateData generic

### LINKED ISSUE #7: Nested structure personalizations
SendGrid supports [structured data and arrays in dynamic templates data](https://sendgrid.com/docs/for-developers/sending-email/using-handlebars/#iterations). It would be great if `Personalization` structure supported this as well. Right now it only supports `[String: String]` as `dynamicTemplateData`. I hope my observation isn't wrong, because I haven't tested SendGrid structured data payload yet. 😄 

I'd suggest to replace it with `[String: Encodable]` or custom `Encodable` structure directly. I've had a look at the source code and it seems the update wouldn't be hard. I'll try to implement it when I have some time.

### PR DESCRIPTION
**Pull request by tonyarnold**
This should allow support for nested Codable structures, and better type support when encoded to JSON (everything needed to be a string, so things like numbers, nil and booleans were not being encoded properly.

It does, sadly, make the declaration of Personalization instances a bit messier. I'm open to approaches here.

Fixes #7

### CODE REVIEW

File: Sources/SendGridKit/Models/Personalization.swift [hunk 1]
```diff
@@ -21,8 +21,8 @@ public struct Personalization: Codable {
     public var substitutions: [String: String]?
     
     /// A collection of key/value pairs following the pattern "key":"value" to substitute handlebar template data
-    public var dynamicTemplateData: [String: String]?
-    
+    public var dynamicTemplateData: DynamicTemplateData?
```
Comment (Andrewangeta) [inline Sources/SendGridKit/Models/Personalization.swift:24]: This technically breaks public API correct? @tonyarnold

Comment (tonyarnold) [reply Sources/SendGridKit/Models/Personalization.swift:24]: Yes, it does, sadly.

Comment (Andrewangeta) [reply Sources/SendGridKit/Models/Personalization.swift:24]: Alright we'll just have new major versions of Kit and the helper library.

Comment (tonyarnold) [reply Sources/SendGridKit/Models/Personalization.swift:24]: Sorry! Without this change, though, it's not really possible to use SendGrid's dynamic templates properly. 

Seems worth it? 😄

**Comment by tonyarnold** [comment]
@Andrewangeta what are the chances of this PR making it into a release (even a tagged beta) at some point soon? Did you need to see any changes to the code?

**Comment by Andrewangeta** [comment]
@tonyarnold Hey! Sorry for the delay. Would you be ok/comfortable taking over and maintaining this repo? I have a TON on my plate constantly and don't want to hinder growth and development of the ecosystem because I'm too busy. I actually don't even use sendgrid myself so I'd much rather someone else with more invested take over and be a contributor and shape the evolution of the package overtime.  Just LMK and you can take the reigns and do what's bets for this library :)

**Comment by tonyarnold** [comment]
@Andrewangeta I'd be happy to help in whatever way makes sense. I'm happy to look after the library for the time being, and if you find yourself with time/inclination in future just let me know.

**Comment by tonyarnold** [comment]
@Andrewangeta are there any expectations, rules, etc around maintaining this package? (and, I assume the parent https://github.com/vapor-community/sendgrid package). 

If I want to make a release, are there docs on how to go about that?

**Review by fpseverino — APPROVED** [review]

**Pull request merged**